gcovr/json_generator.py

```diff
@@ -172,6 +172,7 @@ def _json_from_line(line):
     json_line['count'] = line.count
     json_line['line_number'] = line.lineno
     json_line['gcovr/noncode'] = line.noncode
+    json_line['gcovr/function'] = line.function
```

Comment:
If adding a new JSON field we should also update deserialization, and possibly bump the version number. In the interest of compression we might also skip this field unless it is set?

But as discussed above, we might not need this field in the first place.

Reply:
I'll remove this field and set always noncode for the function line as mentioned above.